### ISSUE TITLE
Unable to invoke any controls or settings

### ISSUE THREAD

**Issue by moertel** [vdesabou/alfred-spotify-mini-player#44]
I'm running Alfred v2.6(374) and have downloaded (from packal.org) and installed the latest mini player workflow. When I start typing, I do see the controls but hitting `enter` does not invoke anything. I enabled Alfred debugging for the workflow but it remains completely silent. Am I missing something? I wasn't even asked for Client ID and Secret on first startup.

![screen shot 2015-01-28 at 11 21 43 pm](https://cloud.githubusercontent.com/assets/1823966/5948360/9dccde94-a744-11e4-8907-a669d385aa66.png)

Edit: Finally seeing something in the debug logs when invoking my assigned hotkey:
```
[ERROR: alfred.workflow.input.scriptfilter] Code 255: Parse error: parse error in /Users/steffi/Library/Application Support/Alfred 2/Alfred.alfredpreferences/workflows/user.workflow.5F010387-DB34-4E25-A09F-888EA9F63AA2/src/workflows.php on line 224
[ERROR: alfred.workflow.input.scriptfilter] XML Parse Error 'The operation couldn’t be completed. (NSXMLParserErrorDomain error 4.)'. Row (null), Col (null): '(null)' in XML:
Parse error: parse error in /Users/steffi/Library/Application Support/Alfred 2/Alfred.alfredpreferences/workflows/user.workflow.5F010387-DB34-4E25-A09F-888EA9F63AA2/src/workflows.php on line 224
```

**Comment by moertel** [comment]
Seems I was able to solve it by replacing the arrays in [line 224](https://github.com/vdesabou/alfred-spotify-mini-player/blob/e16b2dc9386238a1083829dd7405a487ebe5af8b/spotify-mini-player/src/workflows.php#L224) and [line 238](https://github.com/vdesabou/alfred-spotify-mini-player/blob/e16b2dc9386238a1083829dd7405a487ebe5af8b/spotify-mini-player/src/workflows.php#L238) of workflows.php with `array(...)`.

I successfully entered Client ID and Secret but am now seeing this in the debug log when invoking `Authenticate to Spotify` in Alfred:
```
[INFO: alfred.workflow.input.scriptfilter] <?xml version="1.0" encoding="UTF-8"?>
<items><item arg="a:16:{i:0;s:0:&quot;&quot;;i:1;s:0:&quot;&quot;;i:2;s:0:&quot;&quot;;i:3;s:0:&quot;&quot;;i:4;s:0:&quot;&quot;;i:5;s:0:&quot;&quot;;i:6;s:0:&quot;&quot;;i:7;s:11:&quot;Oauth_Login&quot;;i:8;s:0:&quot;&quot;;i:9;s:0:&quot;&quot;;i:10;s:0:&quot;&quot;;i:11;s:0:&quot;&quot;;i:12;s:0:&quot;&quot;;i:13;s:0:&quot;&quot;;i:14;s:0:&quot;&quot;;i:15;s:0:&quot;&quot;;}" valid="yes"><arg>a:16:{i:0;s:0:"";i:1;s:0:"";i:2;s:0:"";i:3;s:0:"";i:4;s:0:"";i:5;s:0:"";i:6;s:0:"";i:7;s:11:"Oauth_Login";i:8;s:0:"";i:9;s:0:"";i:10;s:0:"";i:11;s:0:"";i:12;s:0:"";i:13;s:0:"";i:14;s:0:"";i:15;s:0:"";}</arg><title>Authenticate to Spotify</title><subtitle>This will start the authentication process</subtitle><subtitle mod="alt">Not Available</subtitle><subtitle mod="cmd">Not Available</subtitle><subtitle mod="shift">Not Available</subtitle><subtitle mod="fn">Not Available</subtitle><subtitle mod="ctrl">Not Available</subtitle><icon>./images/settings.png</icon><text></text></item></items>
<?xml version="1.0" encoding="UTF-8"?>
<items><item arg="a:16:{i:0;s:0:&quot;&quot;;i:1;s:0:&quot;&quot;;i:2;s:0:&quot;&quot;;i:3;s:0:&quot;&quot;;i:4;s:0:&quot;&quot;;i:5;s:0:&quot;&quot;;i:6;s:0:&quot;&quot;;i:7;s:11:&quot;Oauth_Login&quot;;i:8;s:0:&quot;&quot;;i:9;s:0:&quot;&quot;;i:10;s:0:&quot;&quot;;i:11;s:0:&quot;&quot;;i:12;s:0:&quot;&quot;;i:13;s:0:&quot;&quot;;i:14;s:0:&quot;&quot;;i:15;s:0:&quot;&quot;;}" valid="yes"><arg>a:16:{i:0;s:0:"";i:1;s:0:"";i:2;s:0:"";i:3;s:0:"";i:4;s:0:"";i:5;s:0:"";i:6;s:0:"";i:7;s:11:"Oauth_Login";i:8;s:0:"";i:9;s:0:"";i:10;s:0:"";i:11;s:0:"";i:12;s:0:"";i:13;s:0:"";i:14;s:0:"";i:15;s:0:"";}</arg><title>Authenticate to Spotify</title><subtitle>This will start the authentication process</subtitle><subtitle mod="alt">Not Available</subtitle><subtitle mod="cmd">Not Available</subtitle><subtitle mod="shift">Not Available</subtitle><subtitle mod="fn">Not Available</subtitle><subtitle mod="ctrl">Not Available</subtitle><icon>./images/settings.png</icon><text></text></item><item arg="a:22:{i:0;s:0:&quot;&quot;;i:1;s:0:&quot;&quot;;i:2;s:0:&quot;&quot;;i:3;s:0:&quot;&quot;;i:4;s:0:&quot;&quot;;i:5;s:0:&quot;&quot;;i:6;s:0:&quot;&quot;;i:7;s:14:&quot;update_library&quot;;i:8;s:0:&quot;&quot;;i:9;s:0:&quot;&quot;;i:10;s:0:&quot;&quot;;i:11;s:0:&quot;&quot;;i:12;s:0:&quot;&quot;;i:13;s:0:&quot;&quot;;i:14;s:0:&quot;&quot;;i:15;s:0:&quot;&quot;;i:16;s:0:&quot;&quot;;i:17;s:0:&quot;&quot;;i:18;i:1;i:19;i:1;i:20;s:0:&quot;&quot;;i:21;s:0:&quot;&quot;;}" valid="yes"><arg>a:22:{i:0;s:0:"";i:1;s:0:"";i:2;s:0:"";i:3;s:0:"";i:4;s:0:"";i:5;s:0:"";i:6;s:0:"";i:7;s:14:"update_library";i:8;s:0:"";i:9;s:0:"";i:10;s:0:"";i:11;s:0:"";i:12;s:0:"";i:13;s:0:"";i:14;s:0:"";i:15;s:0:"";i:16;s:0:"";i:17;s:0:"";i:18;i:1;i:19;i:1;i:20;s:0:"";i:21;s:0:"";}</arg><title>Create library</title><subtitle>when done you'll receive a notification. you can check progress by invoking the workflow again</subtitle><icon>./images/update.png</icon><text></text></item></items>
[ERROR: alfred.workflow.input.scriptfilter] XML Parse Error 'The operation couldn’t be completed. (NSXMLParserErrorDomain error 64.)'. Row (null), Col (null): '(null)' in XML:
<?xml version="1.0" encoding="UTF-8"?>
<items><item arg="a:16:{i:0;s:0:&quot;&quot;;i:1;s:0:&quot;&quot;;i:2;s:0:&quot;&quot;;i:3;s:0:&quot;&quot;;i:4;s:0:&quot;&quot;;i:5;s:0:&quot;&quot;;i:6;s:0:&quot;&quot;;i:7;s:11:&quot;Oauth_Login&quot;;i:8;s:0:&quot;&quot;;i:9;s:0:&quot;&quot;;i:10;s:0:&quot;&quot;;i:11;s:0:&quot;&quot;;i:12;s:0:&quot;&quot;;i:13;s:0:&quot;&quot;;i:14;s:0:&quot;&quot;;i:15;s:0:&quot;&quot;;}" valid="yes"><arg>a:16:{i:0;s:0:"";i:1;s:0:"";i:2;s:0:"";i:3;s:0:"";i:4;s:0:"";i:5;s:0:"";i:6;s:0:"";i:7;s:11:"Oauth_Login";i:8;s:0:"";i:9;s:0:"";i:10;s:0:"";i:11;s:0:"";i:12;s:0:"";i:13;s:0:"";i:14;s:0:"";i:15;s:0:"";}</arg><title>Authenticate to Spotify</title><subtitle>This will start the authentication process</subtitle><subtitle mod="alt">Not Available</subtitle><subtitle mod="cmd">Not Available</subtitle><subtitle mod="shift">Not Available</subtitle><subtitle mod="fn">Not Available</subtitle><subtitle mod="ctrl">Not Available</subtitle><icon>./images/settings.png</icon><text></text></item></items>
<?xml version="1.0" encoding="UTF-8"?>
<items><item arg="a:16:{i:0;s:0:&quot;&quot;;i:1;s:0:&quot;&quot;;i:2;s:0:&quot;&quot;;i:3;s:0:&quot;&quot;;i:4;s:0:&quot;&quot;;i:5;s:0:&quot;&quot;;i:6;s:0:&quot;&quot;;i:7;s:11:&quot;Oauth_Login&quot;;i:8;s:0:&quot;&quot;;i:9;s:0:&quot;&quot;;i:10;s:0:&quot;&quot;;i:11;s:0:&quot;&quot;;i:12;s:0:&quot;&quot;;i:13;s:0:&quot;&quot;;i:14;s:0:&quot;&quot;;i:15;s:0:&quot;&quot;;}" valid="yes"><arg>a:16:{i:0;s:0:"";i:1;s:0:"";i:2;s:0:"";i:3;s:0:"";i:4;s:0:"";i:5;s:0:"";i:6;s:0:"";i:7;s:11:"Oauth_Login";i:8;s:0:"";i:9;s:0:"";i:10;s:0:"";i:11;s:0:"";i:12;s:0:"";i:13;s:0:"";i:14;s:0:"";i:15;s:0:"";}</arg><title>Authenticate to Spotify</title><subtitle>This will start the authentication process</subtitle><subtitle mod="alt">Not Available</subtitle><subtitle mod="cmd">Not Available</subtitle><subtitle mod="shift">Not Available</subtitle><subtitle mod="fn">Not Available</subtitle><subtitle mod="ctrl">Not Available</subtitle><icon>./images/settings.png</icon><text></text></item><item arg="a:22:{i:0;s:0:&quot;&quot;;i:1;s:0:&quot;&quot;;i:2;s:0:&quot;&quot;;i:3;s:0:&quot;&quot;;i:4;s:0:&quot;&quot;;i:5;s:0:&quot;&quot;;i:6;s:0:&quot;&quot;;i:7;s:14:&quot;update_library&quot;;i:8;s:0:&quot;&quot;;i:9;s:0:&quot;&quot;;i:10;s:0:&quot;&quot;;i:11;s:0:&quot;&quot;;i:12;s:0:&quot;&quot;;i:13;s:0:&quot;&quot;;i:14;s:0:&quot;&quot;;i:15;s:0:&quot;&quot;;i:16;s:0:&quot;&quot;;i:17;s:0:&quot;&quot;;i:18;i:1;i:19;i:1;i:20;s:0:&quot;&quot;;i:21;s:0:&quot;&quot;;}" valid="yes"><arg>a:22:{i:0;s:0:"";i:1;s:0:"";i:2;s:0:"";i:3;s:0:"";i:4;s:0:"";i:5;s:0:"";i:6;s:0:"";i:7;s:14:"update_library";i:8;s:0:"";i:9;s:0:"";i:10;s:0:"";i:11;s:0:"";i:12;s:0:"";i:13;s:0:"";i:14;s:0:"";i:15;s:0:"";i:16;s:0:"";i:17;s:0:"";i:18;i:1;i:19;i:1;i:20;s:0:"";i:21;s:0:"";}</arg><title>Create library</title><subtitle>when done you'll receive a notification. you can check progress by invoking the workflow again</subtitle><icon>./images/update.png</icon><text></text></item></items>
```

**Comment by vdesabou** [comment]
> Seems I was able to solve it by replacing the arrays in line 224 and line 238 of workflows.php with array(...).

Looks like you're a developer...do you know why you have to do this?
Are you using a specific PHP version?

I have:

```
php --version
PHP 5.5.14 (cli) (built: Sep  9 2014 19:09:25)
Copyright (c) 1997-2014 The PHP Group
Zend Engine v2.5.0, Copyright (c) 1998-2014 Zend Technologies
```
```
[INFO: alfred.workflow.input.scriptfilter] <?xml version="1.0" encoding="UTF-8"?>
<items><item arg="a:16:{i:0;s:0:&quot;&quot;;i:1;s:0:&quot;&quot;;i:2;s:0:&quot;&quot;;i:3;s:0:&quot;&quot;;i:4;s:0:&quot;&quot;;i:5;s:0:&quot;&quot;;i:6;s:0:&quot;&quot;;i:7;s:11:&quot;Oauth_Login&quot;;i:8;s:0:&quot;&quot;;i:9;s:0:&quot;&quot;;i:10;s:0:&quot;&quot;;i:11;s:0:&quot;&quot;;i:12;s:0:&quot;&quot;;i:13;s:0:&quot;&quot;;i:14;s:0:&quot;&quot;;i:15;s:0:&quot;&quot;;}" valid="yes"><arg>a:16:{i:0;s:0:"";i:1;s:0:"";i:2;s:0:"";i:3;s:0:"";i:4;s:0:"";i:5;s:0:"";i:6;s:0:"";i:7;s:11:"Oauth_Login";i:8;s:0:"";i:9;s:0:"";i:10;s:0:"";i:11;s:0:"";i:12;s:0:"";i:13;s:0:"";i:14;s:0:"";i:15;s:0:"";}</arg><title>Authenticate to Spotify</title><subtitle>This will start the authentication process</subtitle><subtitle mod="alt">Not Available</subtitle><subtitle mod="cmd">Not Available</subtitle><subtitle mod="shift">Not Available</subtitle><subtitle mod="fn">Not Available</subtitle><subtitle mod="ctrl">Not Available</subtitle><icon>./images/settings.png</icon><text></text></item></items>
<?xml version="1.0" encoding="UTF-8"?>
<items><item arg="a:16:{i:0;s:0:&quot;&quot;;i:1;s:0:&quot;&quot;;i:2;s:0:&quot;&quot;;i:3;s:0:&quot;&quot;;i:4;s:0:&quot;&quot;;i:5;s:0:&quot;&quot;;i:6;s:0:&quot;&quot;;i:7;s:11:&quot;Oauth_Login&quot;;i:8;s:0:&quot;&quot;;i:9;s:0:&quot;&quot;;i:10;s:0:&quot;&quot;;i:11;s:0:&quot;&quot;;i:12;s:0:&quot;&quot;;i:13;s:0:&quot;&quot;;i:14;s:0:&quot;&quot;;i:15;s:0:&quot;&quot;;}" valid="yes"><arg>a:16:{i:0;s:0:"";i:1;s:0:"";i:2;s:0:"";i:3;s:0:"";i:4;s:0:"";i:5;s:0:"";i:6;s:0:"";i:7;s:11:"Oauth_Login";i:8;s:0:"";i:9;s:0:"";i:10;s:0:"";i:11;s:0:"";i:12;s:0:"";i:13;s:0:"";i:14;s:0:"";i:15;s:0:"";}</arg><title>Authenticate to Spotify</title><subtitle>This will start the authentication process</subtitle><subtitle mod="alt">Not Available</subtitle><subtitle mod="cmd">Not Available</subtitle><subtitle mod="shift">Not Available</subtitle><subtitle mod="fn">Not Available</subtitle><subtitle mod="ctrl">Not Available</subtitle><icon>./images/settings.png</icon><text></text></item><item arg="a:22:{i:0;s:0:&quot;&quot;;i:1;s:0:&quot;&quot;;i:2;s:0:&quot;&quot;;i:3;s:0:&quot;&quot;;i:4;s:0:&quot;&quot;;i:5;s:0:&quot;&quot;;i:6;s:0:&quot;&quot;;i:7;s:14:&quot;update_library&quot;;i:8;s:0:&quot;&quot;;i:9;s:0:&quot;&quot;;i:10;s:0:&quot;&quot;;i:11;s:0:&quot;&quot;;i:12;s:0:&quot;&quot;;i:13;s:0:&quot;&quot;;i:14;s:0:&quot;&quot;;i:15;s:0:&quot;&quot;;i:16;s:0:&quot;&quot;;i:17;s:0:&quot;&quot;;i:18;i:1;i:19;i:1;i:20;s:0:&quot;&quot;;i:21;s:0:&quot;&quot;;}" valid="yes"><arg>a:22:{i:0;s:0:"";i:1;s:0:"";i:2;s:0:"";i:3;s:0:"";i:4;s:0:"";i:5;s:0:"";i:6;s:0:"";i:7;s:14:"update_library";i:8;s:0:"";i:9;s:0:"";i:10;s:0:"";i:11;s:0:"";i:12;s:0:"";i:13;s:0:"";i:14;s:0:"";i:15;s:0:"";i:16;s:0:"";i:17;s:0:"";i:18;i:1;i:19;i:1;i:20;s:0:"";i:21;s:0:"";}</arg><title>Create library</title><subtitle>when done you'll receive a notification. you can check progress by invoking the workflow again</subtitle><icon>./images/update.png</icon><text></text></item></items>
```


Looks like ```<?xml version="1.0" encoding="UTF-8"?>``` appears twice, that must be the problem for Alfred to parse it..

But again, I have no idea how you could get these kinds of errors..

Let me know if you have anything specific in your environment related to PHP

Thanks

**Comment by moertel** [comment]
Many thanks for the reply. Yep, I'm still on PHP 5.3.26 and [according to the documentation](http://php.net/manual/en/language.types.array.php) that's indeed why the array failed:
> As of PHP 5.4 you can also use the short array syntax, which replaces array() with [].

Good catch! I'll update to a 5.5-ish version and see if that fixes the issues. Probably worth adding this as a system requirement to the documentation.

**Comment by vdesabou** [comment]
> Good catch! I'll update to a 5.5-ish version and see if that fixes the issues. Probably worth adding this as a system requirement to the documentation. 

I think it is easier for me to update the workflow. Can you try this version? https://cloudup.com/cAm6UMGO-ME 
Let me know if you have other issues.

**Comment by moertel** [comment]
The version you uploaded works fine, thanks. The XML parsing problem remains, however. I just installed PHP 5.5 to make sure but the XML error in the debug output looks exactly the same. Any ideas how to troubleshoot this?

**Comment by vdesabou** [comment]
turns out I also have same debug output:

```
[ERROR: alfred.workflow.input.scriptfilter] XML Parse Error 'The operation couldn’t be completed. (NSXMLParserErrorDomain error 64.)'. Row (null), Col (null): 'XML declaration allowed only at the start of the document' in XML:
<?xml version="1.0" encoding="UTF-8"?>
<items><item arg="" valid="no"><arg></arg><title>Your Application Client ID is missing</title><subtitle>Get it from your Spotify Application and copy/paste it here</subtitle><icon>./images/settings.png</icon><text></text></item><item arg="a:16:{i:0;s:0:&quot;&quot;;i:1;s:0:&quot;&quot;;i:2;s:0:&quot;&quot;;i:3;s:0:&quot;&quot;;i:4;s:0:&quot;&quot;;i:5;s:0:&quot;&quot;;i:6;s:68:&quot;Open▹https://developer.spotify.com/my-applications/#!/applications&quot;;i:7;s:0:&quot;&quot;;i:8;s:0:&quot;&quot;;i:9;s:0:&quot;&quot;;i:10;s:0:&quot;&quot;;i:11;s:0:&quot;&quot;;i:12;s:0:&quot;&quot;;i:13;s:0:&quot;&quot;;i:14;s:0:&quot;&quot;;i:15;s:0:&quot;&quot;;}" valid="yes"><arg>a:16:{i:0;s:0:"";i:1;s:0:"";i:2;s:0:"";i:3;s:0:"";i:4;s:0:"";i:5;s:0:"";i:6;s:68:"Open▹https://developer.spotify.com/my-applications/#!/applications";i:7;s:0:"";i:8;s:0:"";i:9;s:0:"";i:10;s:0:"";i:11;s:0:"";i:12;s:0:"";i:13;s:0:"";i:14;s:0:"";i:15;s:0:"";}</arg><title>Open Spotify Application page to get required information</title><subtitle>This will open the Application page with your default browser</subtitle><icon>./images/spotify.png</icon><text></text></item></items>
<?xml version="1.0" encoding="UTF-8"?>
<items><item arg="" valid="no"><arg></arg><title>Your Application Client ID is missing</title><subtitle>Get it from your Spotify Application and copy/paste it here</subtitle><icon>./images/settings.png</icon><text></text></item><item arg="a:16:{i:0;s:0:&quot;&quot;;i:1;s:0:&quot;&quot;;i:2;s:0:&quot;&quot;;i:3;s:0:&quot;&quot;;i:4;s:0:&quot;&quot;;i:5;s:0:&quot;&quot;;i:6;s:68:&quot;Open▹https://developer.spotify.com/my-applications/#!/applications&quot;;i:7;s:0:&quot;&quot;;i:8;s:0:&quot;&quot;;i:9;s:0:&quot;&quot;;i:10;s:0:&quot;&quot;;i:11;s:0:&quot;&quot;;i:12;s:0:&quot;&quot;;i:13;s:0:&quot;&quot;;i:14;s:0:&quot;&quot;;i:15;s:0:&quot;&quot;;}" valid="yes"><arg>a:16:{i:0;s:0:"";i:1;s:0:"";i:2;s:0:"";i:3;s:0:"";i:4;s:0:"";i:5;s:0:"";i:6;s:68:"Open▹https://developer.spotify.com/my-applications/#!/applications";i:7;s:0:"";i:8;s:0:"";i:9;s:0:"";i:10;s:0:"";i:11;s:0:"";i:12;s:0:"";i:13;s:0:"";i:14;s:0:"";i:15;s:0:"";}</arg><title>Open Spotify Application page to get required information</title><subtitle>This will open the Application page with your default browser</subtitle><icon>./images/spotify.png</icon><text></text></item><item arg="a:22:{i:0;s:0:&quot;&quot;;i:1;s:0:&quot;&quot;;i:2;s:0:&quot;&quot;;i:3;s:0:&quot;&quot;;i:4;s:0:&quot;&quot;;i:5;s:0:&quot;&quot;;i:6;s:0:&quot;&quot;;i:7;s:14:&quot;update_library&quot;;i:8;s:0:&quot;&quot;;i:9;s:0:&quot;&quot;;i:10;s:0:&quot;&quot;;i:11;s:0:&quot;&quot;;i:12;s:0:&quot;&quot;;i:13;s:0:&quot;&quot;;i:14;s:0:&quot;&quot;;i:15;s:0:&quot;&quot;;i:16;s:0:&quot;&quot;;i:17;s:0:&quot;&quot;;i:18;i:1;i:19;i:1;i:20;s:0:&quot;&quot;;i:21;s:0:&quot;&quot;;}" valid="yes"><arg>a:22:{i:0;s:0:"";i:1;s:0:"";i:2;s:0:"";i:3;s:0:"";i:4;s:0:"";i:5;s:0:"";i:6;s:0:"";i:7;s:14:"update_library";i:8;s:0:"";i:9;s:0:"";i:10;s:0:"";i:11;s:0:"";i:12;s:0:"";i:13;s:0:"";i:14;s:0:"";i:15;s:0:"";i:16;s:0:"";i:17;s:0:"";i:18;i:1;i:19;i:1;i:20;s:0:"";i:21;s:0:"";}</arg><title>Create library</title><subtitle>when done you'll receive a notification. you can check progress by invoking the workflow again</subtitle><icon>./images/update.png</icon><text></text></item></items>
```

However it does not prevent the Alfred window to open. Let me investigate this..

**Comment by vdesabou** [comment]
I fixed the problem, can you try this version https://cloudup.com/cLMfprJKuUS 

If it still does not work, can you follow steps [here](http://alfred-spotify-mini-player.com/articles/support/) in order to send me TGZ file.

**Comment by moertel** [comment]
That was fast, thanks. The error is now gone for me, too. When invoking "Authenticate to Spotify", it tries to open http://localhost:15298/ in my browser but there does not seem to be a server running on that port. No output in the logs either. Is the TGZ helpful for troubleshooting this?

**Comment by vdesabou** [comment]
hmm..I start the server like this:-

```
else if ($other_action == "Oauth_Login") {
        $cache_log = $w->cache() . '/spotify_mini_player_web_server.log';
        exec("php -S localhost:15298 > \"$cache_log\" 2>&1 &");
        sleep(2);
        exec("open http://localhost:15298");
        return;
    } 
```
So maybe there is a problem with PHP < 5.4 ?

Yes the TGZ file would help as output of command is saved in spotify_mini_player_web_server.log

**Comment by moertel** [comment]
Yes, seems to be related to my PHP version. I replaced the execution with `exec("/usr/local/bin/php -S localhost:15298 > \"$cache_log\" 2>&1 &");` (where Homebrew put my PHP 5.5 installation) and now the server started up just fine. [5.4 just doesn't have it](http://php.net/manual/en/features.commandline.webserver.php):
>  As of PHP 5.4.0, the CLI SAPI provides a built-in web server.

**Comment by vdesabou** [comment]
Ok good, so I must set PHP 5.5 as a required dependency I guess...
So you must be now creating the library?

Note that the version you're using is not Packal friendly, so you'll have a warning once per 24 hour when it will try to auto update. Just make sure to download from Packal when next version is released (likely a 6.0.1 soon)

**Comment by vdesabou** [comment]
>Note that the version you're using is not Packal friendly, so you'll have a warning once per 24 hour when it will try to auto update. Just make sure to download from Packal when next version is released (likely a 6.0.1 soon)

FYI, 6.0.1 is now available, re-download it from the [website](http://alfred-spotify-mini-player.com) and install it in order not to have Packal warning when checking for updates

**Comment by moertel** [comment]
Thanks again for your help. I ended up removing 5.3 completely because even with the 6.0.1 version of your workflow and a working Spotify token I would get an error from the database that table `artists` didn't exist, so whatever PHP5.3's SQLite version is, it's likely incompatible as well.

However, with PHP5.5 everything's working smoothly now. I owe you a :beer: :)

**Comment by vdesabou** [comment]
>Thanks again for your help. I ended up removing 5.3 completely because even with the 6.0.1 version of your workflow and a working Spotify token I would get an error from the database that table artists didn't exist, so whatever PHP5.3's SQLite version is, it's likely incompatible as well

No it is not PHP version related, that was an issue with 6.0 which has been fixed in 6.0.1, but I forgot to tell you to delete two DB files before re-creating the library..That's what you got this error. See details [here](http://alfred-spotify-mini-player.com/blog/Version-6.0.1/)

Anyway, I'm glad it is working now!

>However, with PHP5.5 everything's working smoothly now. I owe you a :beer: :)

Yeah! I received your donation, thanks a lot. I'll buy a beer no later than tonight! Really appreciated ! :smile:
